src/base/misc/fatfs.c

```diff
@@ -1577,7 +1577,8 @@ void build_boot_blk(fatfs_t *f, unsigned char *b)
     case OLDMSD_D:
       txfr->ofs = 0x0000;
       txfr->seg = 0x0070;
-      txfr->ax  = d_o >> 16;
+      txfr->ax  = 0x0200;  /* Magic value which makes MS-DOS 3.10 happy, see */
```

Comment:
Based on your disasm I don't see where does
0x200 come from. I would expect a much smaller
value: number of sector where msdos.sys starts.